### PR TITLE
Fix tests and implement upserts

### PR DESCRIPTION
**Pull request by lilsweetcaligula**
The tests will fail until a new release of the `seneca-store-test` package is made, which does not automatically include "merge:false" tests for SQL databases.

TODO:
- [x] set up the CI
- [x] tidy up
- [x] remove dependency on `lodash`
- [x] The `ids: [...]` directive should be `id: […]`
- [x] Implement `id: [...]` for the `#load` method.

### CODE REVIEW

**Comment by lilsweetcaligula** [comment]
_QUESTION:_ Currently, the store has tests in place for the `toColumnName`/`fromColumnName` options. These options are only tested inside the `#list$` method, and what these options do is - if your column names are, for example, in the `under_score` case, but you want to use `camelCase` for fields in the code, you can achieve that like this:

```javascript
app.use('postgres-store', {
  toColumnName: camelToSnakeCase,
  fromColumnName: snakeToCamelCase
})
```  
  
```javascript
/* Assume, the `foos` table does not have a `barFoo` column - it has, however, the `bar_foo`
 * column. But the aforementioned feature converts the field name to a column name
 * automatically, as seen in the test below:
 */
it('', async () => {
  const foos = await app.make('foos').list$({ barFoo: 'hello' })

  expect(foos.length).to.equal(1)
  expect(foos[0].barFoo).to.equal('hello')
})

```

Given that the feature currently only works in `#list$`, - is this something we want to keep supporting, or should we drop the feature from the newer versions of the `postgres-store` altogether?

If we choose to continue support, do we extend this feature to other methods as well, e.g. `save$`, `remove$`, `load$`? If we choose to continue support, what about table/collection names - if one can have support for columns like that, the developer would surely expect such feature to work on table/collection names as well?

File: lib/intern.js [hunk 1]
```diff
@@ -0,0 +1,494 @@
...
+const Uuid = require('uuid')
+
+
+const intern = {
+  asyncMethod(f) {
+    return function (msg, done) {
+      const seneca = this
+      const p = f.call(seneca, msg)
+
+      Assert('function' === typeof p.then &&
```
Comment (rjrodger) [inline lib/intern.js:13]: this is checking for a Promise object "duck type", not an async function - is that intentional?

File: lib/intern.js [hunk 2]
```diff
@@ -0,0 +1,494 @@
...
+      return p
+        .then(result => done(null, result))
+        .catch(done)
+    }
+  },
+
+
+  compact(obj) {
+    return Object.keys(obj)
+      .map(k => [k, obj[k]])
```
Comment (rjrodger) [inline lib/intern.js:26]: Object.entries would suffice?

File: lib/intern.js [hunk 3]
```diff
@@ -0,0 +1,494 @@
...
+      client.release()
+    }
+
+    return result
+  },
+
+
+  async execQuery(query, ctx) {
+    if (!query) {
+      throw new Error('An empty query is not a valid query')
```
Comment (rjrodger) [inline lib/intern.js:52]: use seneca.fail

File: lib/qbuilder.js [hunk 4]
```diff
@@ -0,0 +1,763 @@
+const Assert = require('assert')
```
Comment (rjrodger) [inline lib/qbuilder.js:1]: is this used only by postgres-store?

File: postgresql-store.js [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+  }
+
+  const store = {
+    name: STORE_NAME,
+
+    close: function (_msg, done) {
+      dbPool.end().then(done).catch(done)
+    },
+
+    save: asyncMethod(async function (msg) {
```
Comment (rjrodger) [inline postgresql-store.js:44]: is seneca-promisify required?

**Review by rjrodger — CHANGES_REQUESTED** [review]

File: lib/intern.js (hunk 1 above)
Comment (lilsweetcaligula) [reply lib/intern.js:13]: Yes, it is intentional, so that custom Promise implementations may be used, if necessary.

File: lib/qbuilder.js (hunk 4 above)
Comment (lilsweetcaligula) [reply lib/qbuilder.js:1]: This flavor of the `QBuilder` is only used by the `postgres-store`. The `mysql-store` uses a different flavor of it. For now, we can think of this module as a set of utilities that are private to the store plugin.

File: postgresql-store.js (hunk 5 above)
Comment (lilsweetcaligula) [reply postgresql-store.js:44]: No, `seneca-promisify` should not be required.

File: lib/intern.js (hunk 2 above)
Comment (lilsweetcaligula) [reply lib/intern.js:26]: Good catch!

**Pull request merged**